### ISSUE TITLE
Minimap feature

### ISSUE THREAD

**Issue by Macumbafeh** [leatrix/leatrix-plus-bcc#73]
Hello,

I just tested the hide addon buttons on minimap feature and Zperl + BankItem minimap buttons are missing.

i added it myself but it could be good to add it for everyone.
The frame names :
Zperl : Zperl_MinimapButton_Frame
![image](https://user-images.githubusercontent.com/47739411/145677575-30c26c8c-1358-4770-adb6-5260de7eeac2.png)

BankItem : BankItems_MinimapButton
![image](https://user-images.githubusercontent.com/47739411/145677591-d4a86a1a-7dc5-484f-8ede-ae74b27c39f1.png)

Maybe there is some others.

**Comment by leatrix** [comment]
Like other minimap button addons, the 'Hide addon buttons' and 'Combine addon buttons' settings work with addons that use LDB buttons - which is almost all of them.    However, a few addons still don't use LDB.

BankItems includes some support for LDB but, strangely, the addon doesn't register the LDB button after defining it.  Open BankItems.lua, search for `tt:AddLine(string.format(L["|cffeda55fAlt-Click` and add three lines near it, as shown below (the three lines start and end with BankItems_Save - I included the preceding lines so you know where to put them).  After making the changes, you can go into BankItems configuration panel and disable the minimap button.  Note that, strangely again, BankItems doesn't include the LDB library with the addon so the button will only show if you have another addon installed which includes the LDB library and loads it before BankItems (most addons which support LDB will include the library).

```
				tt:AddLine(string.format(L["|cffeda55fAlt-Click|r to toggle BankItems %s"], REAGENT_BANK));
			end,
		})

		BankItems_Save["hide"] = false
		local icon = LibStub("LibDBIcon-1.0", true)
		icon:Register("BankItems", BankItemsLauncher, BankItems_Save)

	end
end
```

ZPerl doesn't include LDB support at all.  

Contact the author of each addon and ask them to fully support LDB.  There's no need for either addon to use a custom button.  Other minimap addons such as SexyMap and BasicMinimap don't hide the custom buttons either.

**Comment by Macumbafeh** [comment]
Got it ty.

**Comment by leatrix** [comment]
Since some authors might not update their addons, I added generic LibDBIcon buttons for BankItems and Z-Perl in 2.5.75.alpha.2 (or later) (and the latest alpha releases for Shadowlands and Classic Era/SoM).  Let me know what you think.  If you use the generic button then you can disable the addon's minimap button.  The generic button will just be a launcher though and won't recognise modified clicks (such as alt or shift clicking).  Which do you prefer to use - the new buttons or the old ones?

**Comment by Macumbafeh** [comment]
I tried the version 2.5.75.alpha.5-bcc but it doesn't work. So i guess they are generic buttons.

I checked and i can disable them yep. Ty anyway !

I just use an other addon to align them like this :)
![image](https://user-images.githubusercontent.com/47739411/146013849-dfa0197c-f8fd-4a2f-93ad-ef86c0e1dd1b.png)

After :)
![image](https://user-images.githubusercontent.com/47739411/146013908-1c1e9e4a-7137-4709-9fe5-ed6ea447a46a.png)

**Comment by leatrix** [comment]
I thought about adding an exclusion for BugSack before so it's something I might add.  What addon makes the non-standard buttons work for you?

**Comment by Macumbafeh** [comment]
> I thought about adding an exclusion for BugSack before so it's something I might add. What addon makes the non-standard buttons work for you? I looked at a popular minimap button tidy-up addon and it runs a script every time your screen refreshes - so if your framerate is 100 frames per second, it runs the script 100 times a second - permanently - just to tidy up a few non-standard addon buttons! That's massively wasteful. I would never add anything like that to LTP.

You are talking about TidyMinimap?
Tbh it's an old addon used in retail tbc and works without error on classic. But i never check the code. I have it here
https://github.com/Macumbatbc/TidyMinimap-TBCC

**Comment by leatrix** [comment]
No it's not that one.  I'll look at it when I get a moment though.

**Comment by Macumbafeh** [comment]
> No it's not that one. I'll look at it when I get a moment though.

Ah i see, the one included in LTP which open a frame with all minimap button in it. I tested it fastly

**Comment by leatrix** [comment]
I don't have time at the moment but I can see how it works.  It checks for minimap children and automatically assigns them buttons regardless of whether they are LibDBIcon buttons or not.  I actually coded something like that a week ago or so but I didn't use it.  Instead, LTP checks for LibDBIcon buttons since all addons should be using that anyway (and most do).  So it felt a bit over the top to check for all buttons when in reality it's only needed for a few non-standard buttons.  I'll look at it again soon when I get time though but can't promise anything.  Addons with a minimap button should be using LibDBIcon.

**Comment by Macumbafeh** [comment]
> I don't have time at the moment but I can see how it works. It checks for minimap children and automatically assigns them buttons regardless of whether they are LibDBIcon buttons or not. I actually coded something like that a week ago or so but I didn't use it. Instead, LTP checks for LibDBIcon buttons since all addons should be using that anyway (and most do). So it felt a bit over the top to check for all buttons when in reality it's only needed for a few non-standard buttons. I'll look at it again soon when I get time though but can't promise anything. Addons with a minimap button should be using LibDBIcon.

Inspired by MBB :).

Btw i use those addon buttons :
![IMG_20211214_203630](https://user-images.githubusercontent.com/47739411/146067618-4e12fdd6-9f7b-4e05-8c1f-5e523f5a802e.jpg)

I removed some other

**Comment by leatrix** [comment]
In 2.5.75.alpha.6, there is a new setting called 'Replace custom buttons'.  If enabled, custom buttons will be replaced by LibDBIcon buttons at login.

From the change log:
> Added a new setting called 'Replace custom buttons'.  Most addons use the standard LibDBIcon library for their minimap buttons but a few addons use custom, non-standard buttons instead.  The new setting will replace any custom buttons with LibDBIcon buttons.  This will solve the problems with custom buttons such as not hiding automatically, not following the minimap shape and not being placed in the button frame.  However, replaced buttons will have a generic texture.

As a test, I installed the top 205 addons from CurseForge.  After I had closed about 50 windows so that I could even see the minimap, I looked at the minimap buttons.  So out of 205 addons, 35 of them use minimap buttons.  Out of those 35, 28 use standard LibDBIcon buttons and 7 use non-standard custom buttons.

With 'Replace custom buttons' enabled, the 7 custom buttons were replaced by LibDBIcon buttons which worked fine with 'Hide addon buttons', 'Combine addon buttons' and 'Square minimap'.  However, the replaced buttons have generic textures.  That said, since only 20% of the top 205 addons with minimap buttons use non-standard buttons, it's likely that the average player only has one or two of those installed (if any at all) so using a generic texture shouldn't be a problem.  Modified clicks and right-clicks work fine with the replaced buttons.

Since the buttons are only replaced at login, you'll need to reload if you use the affected addon's configuration panel to toggle the non-standard minimap button in order for the replaced button to be updated.

The 7 addons (out of the top 205 addons) that use non-standard, custom buttons are:

- ATT Classic
- Altoholic
- BankItems
- BtWQuests
- ZPerl
- LFG Bulletin Board
- WIM

If you use any of these addons, please ask the author to use the standard LibDBIcon library for their minimap buttons.

https://www.curseforge.com/wow/addons/leatrix-plus-bcc/files/all

**Comment by leatrix** [comment]
In the latest alpha, there is a setting to exclude BugSack too so that the BugSack icon always remains visible.

**Comment by leatrix** [comment]
In 2.5.75.alpha.10, non-standard buttons should have textures (as long as 'Replace non-standard buttons' is enabled).  I tested it with the 7 non-standard addon buttons (listed above) and all of them had textures.  The only exception is Method Raid Tools because of it's fake LibDBIcon button.

**Comment by leatrix** [comment]
> 
> You are talking about TidyMinimap? Tbh it's an old addon used in retail tbc and works without error on classic. But i never check the code. I have it here https://github.com/Macumbatbc/TidyMinimap-TBCC

It's MBB that has the permanent update script.  On my system, the update script for MBB runs 110 times every single second - permanently - which is ludicrous for a simple minimap button frame.  110 updates every second, or 6600 updates every minute, or 396,000 updates every hour just to put buttons in a frame.

By comparison, LTP has its own button frame which you can enable with 'Combine addon buttons'.  It doesn't run any background code at all until you right-click the minimap and even then it only runs enough code to draw the button frame then it goes back to doing nothing at all again.

**Comment by Hillninja** [comment]
I started using the Combine addon buttons but now i get up a ton of "QuestieFrame170" on the list of my addons: Loved the feature but when i get 3-6 of those it becomes a mess xD

**Comment by leatrix** [comment]
Many thanks for the report about Questie.  It's fixed in 2.5.76 which will be available anytime now.  Any other issues, just let me know.
